compiler/rustc_mir_build/src/check_tail_calls.rs

```diff
@@ -118,21 +118,7 @@ impl<'tcx> TailCallCkVisitor<'_, 'tcx> {
         }
 
         if caller_sig.inputs_and_output != callee_sig.inputs_and_output {
```

Comment:
please also add a test that we anonymize bound vars: `for<'a> fn(&'a ())` and `for<'b> fn(&'b ())`